include/RouteMap.h

```diff
@@ -305,57 +324,119 @@ struct RouteMapPosition {
 };
 
 
+/* Configuration parameters for a route between two points. */
 struct RouteMapConfiguration {
     RouteMapConfiguration () : StartLon(0), EndLon(0), 
           grib(nullptr), grib_is_data_deficient(false) {} /* avoid waiting forever in update longitudes */
     bool Update();
 
     wxString RouteGUID;       /* Route GUID if any */
-    wxString Start;
+    wxString Start;           /* The name of starting position, which is resolved to StartLat/StartLon. */
     wxString StartGUID;
 
-    wxString End;
+    wxString End;             /* The name of the destination position, which is resolved to EndLat/EndLon. */
     wxString EndGUID;
 
-    wxDateTime StartTime;
+    wxDateTime StartTime;     /* The time when the boat leaves the starting position. */
 
     double DeltaTime; /* default time in seconds between propagations */
     double UsedDeltaTime; /* time in seconds between propagations */
 
-    Boat boat;
-    wxString boatFileName;
+    Boat boat;                /* The polars of the boat, used for the route calculation. */
+    wxString boatFileName;    /* The name of the boat XML file referencing polars. */
     
     enum IntegratorType { NEWTON, RUNGE_KUTTA } Integrator;
 
-    double MaxDivertedCourse, MaxCourseAngle, MaxSearchAngle, MaxTrueWindKnots, MaxApparentWindKnots;
-    double MaxSwellMeters, MaxLatitude, TackingTime, WindVSCurrent;
+    // The maximum angle the boat can be diverted from the bearing to the destination,
+    // at each step of the route calculation, in degrees.
+    // This represents the angle away from Start to End bearing (StartEndBearing).
+    // The normal setting is 100 degrees, which speeds up calculations.
+    // If the route needs to go around land, islands or peninsulas, the user can increase the value.
+    // E.g. the boat may have to go in the opposite direction then back to the destination bearing.
+    double MaxDivertedCourse;
```

Comment:
AFAIK, the MaxDivertedCourse is compared against StartEndBearing , which is just the initial bearing. It would make more sense to compare the bearing to the destination at each step of the route calculation.
For example, a route going roughly Baja California to Japan from will have an initial bearing of 316, gradually decreasing to 246.
At the beginning of the route, a MaxDivertedCourse value of 100 degrees would be compared against the initial bearing (316). At the end of the route, the MaxDivertedCourse value would still be compared against the initial bearing (316), even though the boat is now heading to 246, so it would tend to skew the diverted route to the right side of the gc.

For most practical route, it may not matter.

Reply:
I'm still reviewing the code and trying to understand how all of these parameters work together.